netutils/dhcpc/dhcpc.c

```diff
@@ -326,41 +326,48 @@ static uint8_t dhcpc_parseoptions(FAR struct dhcpc_state *presult,
 
             /* Get subnet mask in network order */
 
-            memcpy(&presult->netmask.s_addr, optptr + 2, 4);
+            if (optptr + 6 <= end)
```

Comment:
Done

Reply:
Which printf should be used then? `ninfo`?
And be aware that this only happens for a malformed request.

Reply:
> Which printf should be used then? `ninfo`?
> And be aware that this only happens for a malformed request.

I think in this case should be nerr(), because it is a critical fault

Reply:
Done